control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -623,6 +623,23 @@ func (r *HostedControlPlaneReconciler) ensureControlPlane(ctx context.Context, h
 		return fmt.Errorf("failed to get pki secret: %w", err)
 	}
 
+	// Generate guest cluster internal network API Server kubeconfig secret for CAPI components.
+	// The client used by CAPI machine controller expects the kubeconfig to follow this naming convention
+	// https://github.com/kubernetes-sigs/cluster-api/blob/5c85a0a01ee44ecf7c8a3c3fdc867a88af87d73c/util/secret/secret.go#L29-L33
+	CAPIKubeConfigSecretRef := &hyperv1.KubeconfigSecretRef{
+		Name: fmt.Sprintf("%s-kubeconfig", hcp.GetName()),
+		Key:  "value",
+	}
+	CAPIKubeconfigSecret, err := generateKubeconfigSecret(hcp.GetNamespace(), CAPIKubeConfigSecretRef, pkiSecret.Data["internal-admin.kubeconfig"])
+	if err != nil {
+		return fmt.Errorf("failed to create kubeconfig secret manifest for management cluster: %w", err)
+	}
+	CAPIKubeconfigSecret.OwnerReferences = ensureHCPOwnerRef(hcp, CAPIKubeconfigSecret.OwnerReferences)
+	if err := r.Create(ctx, CAPIKubeconfigSecret); err != nil && !apierrors.IsAlreadyExists(err) {
```

Comment:
The secret should be managed with a CreateOrUpdate call so that we're reconciling